### ISSUE TITLE
reducer() is not compatible with Redux reducer spec

### ISSUE THREAD

**Issue by osak** [cartant/ts-action#23]
Following code does not compile with `--strict` option:
```typescript
import {action, reducer, on} from 'ts-action';
import {combineReducers, createStore} from 'redux';

const A = action('a');

interface State {}

const rootReducer = reducer<State>([
    on(A, (s, a) => s)
], {});

createStore(rootReducer);
```
with error:
```
% npx tsc --version
Version 2.9.2
% npx tsc --strict a.ts
a.ts:12:13 - error TS2345: Argument of type 'Reducer<State>' is not assignable to parameter of type 'Reducer<State, Action<string>>'.
  Types of parameters 'state' and 'state' are incompatible.
    Type 'State | undefined' is not assignable to type 'State'.
      Type 'undefined' is not assignable to type 'State'.

12 createStore(rootReducer);
               ~~~~~~~~~~~
```
Apparently [Redux's `Reducer` is designed to accept `undefined` as state](https://github.com/reduxjs/redux/blob/master/index.d.ts#L59), as explained in [the official document](https://redux.js.org/basics/reducers#handling-actions), but `ts-action`'s `Reducer` does not.

**Comment by cartant** [comment]
Yep, the `state` parameter in the [`Reducer` type](https://github.com/cartant/ts-action/blob/v6.0.2/source/reducer.ts#L8) should support `undefined` being passed:

```ts
export type Reducer<S> = (state: S | undefined, action: Action<string>) => S;
                                 ~~~~~~~~~~~~~
```

Well spotted. Thanks.

**Comment by cartant** [comment]
Should be fixed in 6.0.3.
